plutus-core/testlib/PlutusCore/Test.hs

```diff
@@ -302,69 +320,57 @@ goldenUPlc ::
   String ->
   a ->
   TestNested
-goldenUPlc = goldenUPlcWith ".uplc" ppThrow
+goldenUPlc = goldenUPlcWith ppCatch
 
 goldenUPlcReadable ::
   (ToUPlc a UPLC.DefaultUni UPLC.DefaultFun) =>
   String ->
   a ->
   TestNested
-goldenUPlcReadable = goldenUPlcWith ".uplc-readable" ppThrowReadable
-
-goldenUPlcCatch ::
-  (ToUPlc a UPLC.DefaultUni UPLC.DefaultFun) =>
-  String ->
-  a ->
-  TestNested
-goldenUPlcCatch = goldenUPlcWith ".uplc-catch" ppCatch
+goldenUPlcReadable = goldenUPlcWith ppCatchReadable
 
 goldenTEval ::
   (ToTPlc a TPLC.DefaultUni TPLC.DefaultFun) =>
   String ->
   [a] ->
   TestNested
 goldenTEval name values =
-  nestedGoldenVsDocM name ".teval" $ prettyPlcClassicDebug <$> (rethrow $ runTPlc values)
+  nestedGoldenVsDocM name ".eval" $ ppCatch $ runTPlc values
 
 goldenUEval ::
   (ToUPlc a TPLC.DefaultUni TPLC.DefaultFun) =>
   String ->
   [a] ->
   TestNested
 goldenUEval name values =
-  nestedGoldenVsDocM name ".ueval" $ prettyPlcClassicDebug <$> (rethrow $ runUPlc values)
-
-goldenTEvalCatch ::
-  (ToTPlc a TPLC.DefaultUni TPLC.DefaultFun) =>
-  String ->
-  [a] ->
-  TestNested
-goldenTEvalCatch name values = nestedGoldenVsDocM name ".teval-catch" $ ppCatch $ runTPlc values
+  nestedGoldenVsDocM name ".eval" $ ppCatch $ runUPlc values
 
-goldenUEvalCatch ::
+goldenUEvalLogs ::
   (ToUPlc a TPLC.DefaultUni TPLC.DefaultFun) =>
   String ->
   [a] ->
   TestNested
-goldenUEvalCatch name values = nestedGoldenVsDocM name ".ueval-catch" $ ppCatch $ runUPlc values
+goldenUEvalLogs name values =
+  nestedGoldenVsDocM name ".eval" $ ppCatch $ runUPlcLogs values
 
--- | Similar to @goldenUEval@ but with profiling turned on.
+-- | This is mostly useful for profiling a test that is normally
+-- tested with one of the other functions, as it's a drop-in
+-- replacement and you can then pass the output into `traceToStacks`.
 goldenUEvalProfile ::
   (ToUPlc a TPLC.DefaultUni TPLC.DefaultFun) =>
   String ->
   [a] ->
   TestNested
 goldenUEvalProfile name values =
-  nestedGoldenVsDocM name ".ueval-profile" $
-    pretty . view _2 <$> (rethrow $ runUPlcProfile values)
+  nestedGoldenVsDocM name ".eval" $ ppCatch $ runUPlcProfile values
 
-goldenUPlcBudget ::
+goldenUEvalBudget ::
   (ToUPlc a TPLC.DefaultUni TPLC.DefaultFun)
   => String
   -> [a]
   -> TestNested
-goldenUPlcBudget name values =
-  nestedGoldenVsDocM name ".budget" $ ppCatch $ view _2 <$> runUPlcBudget values
+goldenUEvalBudget name values =
+  nestedGoldenVsDocM name ".eval" $ ppCatch $ runUPlcBudget values
```

Comment:
Any particular reason to use `.eval` for all these cases rather than differentiating them?

Reply:
Is it useful? It's unclear that the suffix helps you at all here! I think it's somewhat useful to distinguish the files containing programs, since you could imagine syntax highlighting or tools that process them. TBH... that makes me wonder if we should just not use an extra suffix for these ones at all :thinking: 

The other advantage of using the same suffix is that if you change the type of output you don't have to remove and recreate the file. e.g. if you want to change a test to also print the logs you can just change the test function.

Reply:
Just a bit more convenient when you want to run different tests (e.g. godlenUEvalBudget and goldenUEvalLogs) on the same code, since you can use the same name.

Reply:
Turns out we had precisely one case of name collision in the repository, which was some typechecking tests, which makes me think it doesn't matter. It's also not hard to change the test name.